Robot2019/src/main/java/frc/robot/commands/Climb.java

```diff
@@ -33,28 +38,30 @@ protected void initialize() {
   // Called repeatedly when this Command is scheduled to run
   @Override
   protected void execute() {
-    dt.drive(-0.5, -0.5);
-    if (up) {
-      climber.runClimber(0.5);
-      if (!climber.needToClimb())
-        up = false;
+    if (state == State.CLIMB) {
+      climb();
+      if (!dt.isStalled() || Math.abs(dtJoy.getY()) > 0.1) { // 0.1 = Placeholder/approximation 
+        state = State.RETRACT;
+        climber.resetEncoder();
+      }
     } else {
-      climber.runClimber(-0.5);
-      if (!climber.canDrop())
-        up = true;
+      retract();
+      if (climber.getEncDistance() > 23) { // 23 = Placeholder/approximation 
+        state = State.FINISHED;
+      }
```

Comment:
This code only goes up and then down once. Is that what we want to do?

Reply:
That part only occurs after the stage moves to RETRACT.